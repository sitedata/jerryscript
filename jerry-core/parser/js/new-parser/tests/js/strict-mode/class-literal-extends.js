/* This test contains class literal parser tests. */

class A extends Array {
  constructor(a) {
    super(b);
    super.f = 6;
    super["g"]();
  }

  func() {
    super.g(5);
    eval('');
  }
};
