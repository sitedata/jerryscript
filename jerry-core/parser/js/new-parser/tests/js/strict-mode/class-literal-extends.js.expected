
--- Script parsing start ---

  [  1] CBC_PUSH_LITERAL idx:1->ident(Array)
  [  1] CBC_EXT_SUPER_CLASS_CREATE_CONTEXT
  [  2] CBC_EXT_PUSH_CLASS_CONSTRUCTOR
  [  3] CBC_CREATE_OBJECT

--- Class constructor parsing start ---

  [  1] CBC_EXT_PUSH_CONSTRUCTOR_SUPER
  [  2] CBC_PUSH_LITERAL idx:1->ident(b)
  [  0] CBC_EXT_SUPER_CALL byte_arg:1
  [  1] CBC_EXT_PUSH_CONSTRUCTOR_SUPER_PROP
  [  1] CBC_EXT_SUPER_PROP_ASSIGN
  [  2] CBC_PUSH_NUMBER_POS_BYTE number:6
  [  0] CBC_ASSIGN_PROP_LITERAL idx:2->string(f)
  [  1] CBC_EXT_PUSH_CONSTRUCTOR_SUPER_PROP
  [  3] CBC_PUSH_PROP_LITERAL_REFERENCE idx:3->string(g)
  [  3] CBC_EXT_SUPER_PROP_CALL
  [  0] CBC_CALL0_PROP
  [  1] CBC_EXT_PUSH_CONSTRUCTOR_THIS
  [  0] CBC_RETURN

Final byte code dump:

  Maximum stack depth: 4
  Flags: [small_lit_enc,strict_mode,constructor]
  Argument range end: 1
  Register range end: 1
  Identifier range end: 2
  Const literal range end: 4
  Literal range end: 4

   0 : CBC_EXT_PUSH_CONSTRUCTOR_SUPER
   2 : CBC_PUSH_LITERAL idx:1(ident)->ident(b)
   4 : CBC_EXT_SUPER_CALL byte_arg:1
   7 : CBC_EXT_PUSH_CONSTRUCTOR_SUPER_PROP
   9 : CBC_EXT_SUPER_PROP_ASSIGN
  11 : CBC_PUSH_NUMBER_POS_BYTE number:6
  13 : CBC_ASSIGN_PROP_LITERAL idx:2(lit)->string(f)
  15 : CBC_EXT_PUSH_CONSTRUCTOR_SUPER_PROP
  17 : CBC_PUSH_PROP_LITERAL_REFERENCE idx:3(lit)->string(g)
  19 : CBC_EXT_SUPER_PROP_CALL
  21 : CBC_CALL0_PROP
  22 : CBC_EXT_PUSH_CONSTRUCTOR_THIS
  24 : CBC_RETURN

Byte code size: 25 bytes

--- Class constructor parsing end ---

  [  3] CBC_EXT_SET_CLASS_LITERAL idx:2->function

--- Function parsing start ---

  [  1] CBC_EXT_PUSH_SUPER
  [  3] CBC_PUSH_PROP_LITERAL_REFERENCE idx:0->string(g)
  [  4] CBC_PUSH_NUMBER_POS_BYTE number:5
  [  4] CBC_EXT_SUPER_PROP_CALL
  [  0] CBC_CALL1_PROP
  [  2] CBC_PUSH_TWO_LITERALS idx:1->ident(eval) idx:2->string()
  [  2] CBC_EXT_CLASS_EVAL byte_arg:2
  [  0] CBC_CALL1

Final byte code dump:

  Maximum stack depth: 4
  Flags: [small_lit_enc,strict_mode,arguments_needed]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 1
  Const literal range end: 3
  Literal range end: 3

   0 : CBC_EXT_PUSH_SUPER
   2 : CBC_PUSH_PROP_LITERAL_REFERENCE idx:1(lit)->string(g)
   4 : CBC_PUSH_NUMBER_POS_BYTE number:5
   6 : CBC_EXT_SUPER_PROP_CALL
   8 : CBC_CALL1_PROP
   9 : CBC_PUSH_TWO_LITERALS idx:0(ident)->ident(eval) idx:2(lit)->string()
  12 : CBC_EXT_CLASS_EVAL byte_arg:2
  15 : CBC_CALL1
  16 : CBC_RETURN_WITH_BLOCK

Byte code size: 17 bytes

--- Function parsing end ---

  [  3] CBC_SET_LITERAL_PROPERTY idx:4->function idx:3->string(func)
  [  3] CBC_EXT_INHERIT_AND_SET_CONSTRUCTOR
  [  2] CBC_SET_PROPERTY idx:5->string(prototype)
  [  1] CBC_ASSIGN_SET_IDENT idx:0->var_ident(A)
  [  0] CBC_CONTEXT_END

Final byte code dump:

  Maximum stack depth: 3
  Flags: [small_lit_enc]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 2
  Const literal range end: 4
  Literal range end: 6

   0 : CBC_DEFINE_VARS from: 0 to: 0
         idx:0(ident)->var_ident(A)
   2 : CBC_PUSH_LITERAL idx:1(ident)->ident(Array)
   4 : CBC_EXT_SUPER_CLASS_CREATE_CONTEXT offset:19(->23)
   7 : CBC_EXT_PUSH_CLASS_CONSTRUCTOR
   9 : CBC_CREATE_OBJECT
  10 : CBC_EXT_SET_CLASS_LITERAL idx:4(lit)->function
  13 : CBC_SET_LITERAL_PROPERTY idx:5(lit)->function idx:2(lit)->string(func)
  16 : CBC_EXT_INHERIT_AND_SET_CONSTRUCTOR
  18 : CBC_SET_PROPERTY idx:3(lit)->string(prototype)
  20 : CBC_ASSIGN_SET_IDENT idx:0(ident)->var_ident(A)
  22 : CBC_CONTEXT_END
  23 : CBC_RETURN_WITH_BLOCK

Byte code size: 24 bytes

Script parsing successfully completed. Total byte code size: 66 bytes

--- Script parsing end ---

